src/tables/colr.rs

```diff
@@ -14,6 +23,139 @@ struct BaseGlyphRecord {
     num_layers: u16,
 }
 
+/// A [ClipBox](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct ClipBox {
+    /// The horizontal minimum of the clip box.
+    pub x_min: f32,
+    /// The vertical minimum of the clip box.
+    pub y_min: f32,
+    /// The horizontal maximum of the clip box.
+    pub x_max: f32,
+    /// The vertical maximum of the clip box.
+    pub y_max: f32,
+}
+
+/// A paint.
+#[derive(Clone, Debug)]
+pub enum Paint<'a> {
+    /// A paint with a solid color.
+    Solid(RgbaColor),
+    /// A paint with a linear gradient.
+    LinearGradient(LinearGradient<'a>),
+    /// A paint with a radial gradient.
+    RadialGradient(RadialGradient<'a>),
+    /// A paint with a sweep gradient.
+    SweepGradient(SweepGradient<'a>),
+}
+
+/// A [clip record](
+/// https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug)]
+struct ClipRecord {
+    /// The first glyph ID for the range covered by this record.
+    pub start_glyph_id: GlyphId,
```

Comment:
No need to be `pub`?

Reply:
Nope, should there be a need to? It's only needed in the `colr` file.

Reply:
As long as it compiles I'm fine. I'm not sure why Rust doesn't warn about pub fields in private structs.